transformers/struct.go

```diff
@@ -151,7 +151,11 @@ func TransformWithStruct(st any, opts ...StructTransformerOption) schema.Transfo
 				}
 			}
 		}
-
+		// PKs can be defined in the struct tags or via the WithPrimaryKeys option, so the only heuristic we can apply
+		// is to validate that the total number of created PKs is not less than the total number of created PKs
+		if len(t.table.PrimaryKeys()) < len(t.pkFields) {
```

Comment:
Nice 🚀 I get the check is a bit tricky as we define PKS on fields, but validate on columns

Reply:
What if one of the primary keys isn't a field on the struct - i.e. it's a custom column - will this check still work? I had a case like that not too long ago, can try and find it if you need an example

Reply:
Ah, right it won't work

Reply:
@hermanschaaf - if you can find the example that would be great!

Reply:
@bbernays Sure, here's one recent example from the Alicloud plugin that came to mind: https://github.com/cloudquery/cloudquery/blob/main/plugins/source/alicloud/resources/services/oss/buckets.go#L10

But I think it will happen any time we put the multiplexer value(s) as separate columns and want to also have them be part of the PK. I believe there are lots of examples of this in AWS as well

Reply:
I think we will need to keep track which PKs matches a column here https://github.com/cloudquery/plugin-sdk/blob/08bace89c708ca7f20490ce9756f8276b7e5d6f2/transformers/struct.go#L271 and fail at the end if we didn't have a match for all of them

Reply:
Updated the code to do this

Reply:
(Still working on the tests)...